documentation/Data-View-Filters.md

```diff
@@ -0,0 +1,385 @@
+
+#### Creating filter data with multiple values
+
+You will have noticed the above example created a comma-delimited string of values. This is the preferred way to represent multi-value data in filter definitions, as it's much easier to create database queries to partially match seperate substrings than it is to search inside arrays. As to how the filters parse these strings, please see the ["delimiter" property below](#options-list).
+
+#### How is Filter data generated?
+
+There is a script, `generateFilterDataFields.ts` that can create and update all filter data columns. The script scans the system for existing filter data columns and updates them or creates new ones based on what's in the `data_view_column_definitions` table. It can also (optionally) regenerate ALL values for existing filter_data fields. (Note that if the column definitions have been removed, the generated "filterData" fields will be removed from the database.)
+
+Currently, the only time the app automatically runs it is as an option on the [modifyRecord](List-of-Action-plugins.md#modify-record) action, which ensures that new records automatically have filter data generated for them.
+
+However, there is an endpoint that can be called as required to regenerate all filter data throughout the system. You would normally want to run this after updating or adding filter data column definitions to ensure all data is up-to-date.
+
+The API is:
+
+```
+/admin/generate-filter-data-fields?<query>
+```
+
+Query parameters:
+
+- `table`: the name of the database table to update filter_data for. If not provided, it will update *all* filter data in the system
+- `fullUpdate` (`true`/`false`): if `true`, the script will re-compute (i.e. will re-evaluate every record) the filter_data for all existing filter data as well as generating values for missing data. If `false`, it will only update records where all filter data columns are `null` (i.e. these should be *new* records). By default, `fullUpdate` is `false` when a `table` is specified (this is what `modifyRecord` uses), but `true` when no table is specified (i.e. all tables).
+
+
+### Filter columns
+
+The columns that will appear as available filters for a given data view are defined in `filter_include_columns` and `filter_exclude_columns`. These work the same way as the table view and detail view [include/exclude lists](Data-View.md#data_view-table), but with the following differences:
+
+- if nothing is specified (i.e. `null` value), the defaults will be the same as the table view columns that are being returned. This is because, by default, we expect the available filters to be for the same data as the table columns.
```

Comment:
Maybe specify the "columns" for completeness

Reply:
Not sure what you mean here -- which "columns" are you referring to?

Reply:
When you say "will be the same as the table view columns" I though you meant a data_base column, so the reason why 'Specify', but I guess this is about the columns displayed to the user on the DataView page?

Reply:
Yeah, as I've mentioned elsewhere, when I talk about "fields", I mean data base columns. When I talk about "columns" I mean data view columns. Usually these are the same, but the distinction is important when they are different. I've tried to be consistent about this, but please point out where I may have slipped.